modules/components/pages/outputs/couchbase.adoc

```diff
@@ -0,0 +1,272 @@
+= couchbase
+:type: output
+:status: experimental
+:categories: ["Integration"]
+
+// © 2024 Redpanda Data Inc.
+
+
+component_type_dropdown::[]
+
+
+Perform operations against a Couchbase database for each message to store or delete data.
+
+Introduced in version 4.37.0.
+
+[tabs]
+======
+Common::
++
+--
+```yml
+# Common configuration fields, showing default values
+output:
+  label: ""
+  couchbase:
+    url: couchbase://localhost:11210 # No default (required)
+    username: "" # No default (optional)
+    password: "" # No default (optional)
+    bucket: "" # No default (required)
+    id: ${! json("id") } # No default (required)
+    content: "" # No default (optional)
+    operation: upsert
+    max_in_flight: 64
+    batching:
+      count: 0
+      byte_size: 0
+      period: ""
+      check: ""
+```
+--
+Advanced::
++
+--
+```yml
+# All configuration fields, showing default values
+output:
+  label: ""
+  couchbase:
+    url: couchbase://localhost:11210 # No default (required)
+    username: "" # No default (optional)
+    password: "" # No default (optional)
+    bucket: "" # No default (required)
+    collection: _default
+    transcoder: legacy
+    timeout: 15s
+    id: ${! json("id") } # No default (required)
+    content: "" # No default (optional)
+    operation: upsert
+    max_in_flight: 64
+    batching:
+      count: 0
+      byte_size: 0
+      period: ""
+      check: ""
+      processors: [] # No default (optional)
+```
+--
+======
+
+== Performance
+
+For improved performance, this output sends multiple messages in parallel. You can tune the maximum number of in-flight messages or message batches, using the `max_in_flight` field.
+
+This output can also send message batches. You can configure batches at both the input and output level. For more information, see xref:configuration:batching.adoc[Message Batching].
```

Comment:
The relationship between 'in-flight messages' and message batches is unclear here.

> You can tune the maximum number of in-flight messages or message batches, using the `max_in_flight` field.

This implies that message batches is a synonym of in-flight messages.

> This output can also send message batches. You can configure batches at both the input and output level. For more information, see xref:configuration:batching.adoc[Message Batching].

Then, this implies that message batches are entirely different. We also don't mention the `batching.*` configuration options here but I think it would be useful.

Reply:
Updated as requested. As in previous PR, I've raised an issue to sort out this content site wide so it is standardized but also contains contextual information where necessary.